### PR TITLE
feat(NODE-4196): add support for showExpandedEvents in change streams

### PR DESCRIPTION
**Pull request by baileympearson**
#### What is changing?

This PR adds support for new change stream events guarded behind the `showExpandedEvents` change stream pipeline stage option.  It also adds new fields to some change events when the same flag is enabled.

This PR also contains changes from the following tickets:

- https://jira.mongodb.org/browse/NODE-4205 (see 03886b192a5f370f2e59c6c056c0250b927ffe18)
- https://jira.mongodb.org/browse/DRIVERS-2322 (see 6546428f64088dce9fb3f40bd794a98b3715e5df)

##### Is there new documentation needed for these changes?
No.

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the correct format: `<type>(NODE-xxxx)<!>: <description>`
- [x] Changes are covered by tests
- [x] New TODOs have a related JIRA ticket

### CODE REVIEW

**Review by nbbeeken — CHANGES_REQUESTED** [review]

File: src/change_stream.ts [hunk 1]
```diff
@@ -358,6 +405,91 @@ export interface ChangeStreamInvalidateDocument extends ChangeStreamDocumentComm
   operationType: 'invalidate';
 }
 
+/**
+ * Only present when the `showExpandedEvents` flag is enabled.
+ * @public
+ * @see https://www.mongodb.com/docs/manual/reference/change-events/#invalidate-event
```
Comment (nbbeeken) [inline src/change_stream.ts:411]: The see URLs need updating

File: src/change_stream.ts [hunk 2]
```diff
@@ -225,13 +239,40 @@ export interface ChangeStreamDocumentCommon {
...
+  /**
+   * The UUID of the collection that the operation was performed on.
+   *
+   * Only present when the `showExpandedEvents` flag is enabled.
+   *
+   * **NOTE:** collectionUUID will only be a UUID when the promoteBuffers flag is to true.
+   *
+   * @since 6.1.0
+   */
+  collectionUUID: Binary;
```
Comment (nbbeeken) [inline src/change_stream.ts:253]:
```
**NOTE:** collectionUUID will only be a UUID when the promoteBuffers flag is to true.
```
I think this doc is out of date now, you mean will be a Binary right?

File: src/change_stream.ts [hunk 3]
```diff
@@ -225,13 +239,40 @@ export interface ChangeStreamDocumentCommon {
   lsid?: ServerSessionId;
 }
 
+/** @public */
+export interface ChangeStreamDocumentCollectionUUID {
+  /**
+   * The UUID of the collection that the operation was performed on.
```
Comment (nbbeeken) [inline src/change_stream.ts:245]:
```suggestion
   * The UUID (Binary subtype 4) of the collection that the operation was performed on.
```

**Review by nbbeeken — CHANGES_REQUESTED** [review]

File: src/change_stream.ts (hunk 1 above)
Comment (baileympearson) [reply src/change_stream.ts]: from slack: the docs aren't ready for us to consume yet.  I created a ticket for us to update these urls to point to the correct event documentation and linked it to the docsp ticket for these changes (https://jira.mongodb.org/browse/NODE-4295).

**Review by nbbeeken — APPROVED** [review]

**Review by dariakp — APPROVED** [review]

**Pull request merged**